Sim/SimG4Components/src/SimG4Svc.cpp

```diff
@@ -64,13 +64,14 @@ StatusCode SimG4Svc::initialize() {
   }
 
   m_runManager.Initialize();
+
+  std::unique_ptr<G4VisManager> visManager(new G4VisExecutive);
+
   if (m_interactiveMode) {
-    G4VisManager* visManager = new G4VisExecutive;
     visManager->Initialize();
     // Define UI terminal for interactive mode
-    G4UIsession* session = new G4UIterminal;
+    std::unique_ptr<G4UIsession> session(new G4UIterminal());
```

Comment:
You could have a look at http://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique

Reply:
Is there any problem using unique_ptr in this case ?

Reply:
Hi Nilou,

no, `make_unique` essentially does the same as the line above, but makes for a clearer syntax ( you never even have to write `new`) and is recommended for a few more technical reasons.